### PR TITLE
WIP fwupd: 1.0.5 -> 1.1.0

### PR DESCRIPTION
**Pull request by Moredread**
We override the ESP mount point in the config file /etc/fwupd/uefi.conf
(available since version 1.0.6), as it is set to a path in the nix store
during build time.

Current issues: See issue #39386 

Also:

Tests are disabled as it needs /etc/os-release, which is not available
when building with sandboxing enabled.

I also get a lot of `bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)` during the build, but am not sure if this is an issue with my system configuration.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
You are running into this problem regarding locales: https://github.com/NixOS/nixpkgs/issues/38991

**Comment by Mic92** [comment]
cc @jtojnar

File: nixos/modules/services/hardware/fwupd.nix [hunk 1]
```diff
@@ -71,6 +71,13 @@ in {
           BlacklistPlugins=${lib.concatStringsSep ";" cfg.blacklistPlugins}
         '';
       };
+      "fwupd/uefi.conf" = {
+        source = pkgs.writeText "daemon.conf" ''
```
Comment (jtojnar) [inline nixos/modules/services/hardware/fwupd.nix]: Why do the names of the files mismatch?

**Comment by jtojnar** [comment]
From the [changelog](https://github.com/hughsie/fwupd/blob/1.7.0/NEWS#L1-L47), the only thing that caught my eye were bash completions, but they use an [exact path](https://github.com/hughsie/fwupd/blob/1.0.7/data/bash-completion/meson.build#L4) so it works for us. Upstream should probably switch to something like the following, as per https://www.bassi.io/articles/2018/03/15/pkg-config-and-paths/.

```meson
bash_completion_dep = dependency('bash_completion')
completionsdir = bash_completion_dep.get_pkgconfig_variable(
  'completionsdir',
  define_variable: [ 'prefix', prefix ],
)
```

@Mic92 I do not think that is #38991, the packages are built in sandbox so there should be no conflict with system archive. I have always seen the warning when setting `LC_ALL`.

@Moredread Regarding `/etc/os-release`, that should be reported upstream, it was already fixed once (by skipping the test IIRC).

Could you add a comment saying that `/etc/fwupd/uefi.conf` is created manually in the NixOS module, next to `filesInstalledToEtc`, so that we have the mismatch documented?

@GrahamcOfBorg test fwupd

**Comment by Moredread** [comment]
@jtojnar I'll try to get to it later today but I'm on vacation the next week or so, so I am not sure when I have the time.

**Comment by Moredread** [comment]
@jtojnar have added the comment and created an upstream issue for the failing test.

**Comment by Moredread** [comment]
@jtojnar the issue for the missing os-release file has been closed upstream as nofix

File: nixos/modules/services/hardware/fwupd.nix [hunk 2]
```diff
@@ -71,6 +71,13 @@ in {
           BlacklistPlugins=${lib.concatStringsSep ";" cfg.blacklistPlugins}
         '';
       };
+      "fwupd/uefi.conf" = {
+        source = pkgs.writeText "uefi.conf" ''
+          [uefi]
+          OverrideESPMountPoint=/boot
```
[the remark this answers is not in the document]
Comment (Moredread) [reply nixos/modules/services/hardware/fwupd.nix]: Done :)

**Comment by Moredread** [comment]
Besides the test issue the PR seems to be working and could be merged IMHO.

**Comment by jtojnar** [comment]
Could you use the fix from https://github.com/NixOS/nixpkgs/issues/40872?

**Comment by Moredread** [comment]
In principle that would work (although I'd prefere to add the /tmp/os-release path to the list).

This would expose the os-release parser to a potential attacker with arbitrary rights on a  misconfigured system without the file in the default location. I'm not sure how much that'd an issue.

**Comment by jtojnar** [comment]
Sounds reasonable.

By the way, 1.0.8 is out: [NEWS](https://github.com/hughsie/fwupd/blob/1.0.8/NEWS)

**Comment by Moredread** [comment]
@jtonar The patch we apply doesn't work there anymore. I'm not sure when I'll have time to look at the issue, but will leave a comment when I work on that.

**Comment by jtojnar** [comment]
Looks like the patch needs to update the context of `po/make-images.sh @@ -7,6 +7,7 @@` from `gzip -f ${x}` to `gzip -fn9 ${x}`.

**Comment by jtojnar** [comment]
I will try to update it. The 1.10 changes look interesting.

**Comment by jtojnar** [comment]
Pushed. It would be great if anyone with relevant hardware could test this.

**Comment by jtojnar** [comment]
Since this is a huge improvement we should probably merge it even if the UEFI update is still broken.

**Comment by Moredread** [comment]
@peterhoeg I had a few problems with permission, sometimes it worked, sometimes it didn't. (See https://github.com/NixOS/nixpkgs/issues/39386)

You also have to copy over the UEFI files for fwupd manually, and remove them after the update has finished.

I did a successful FW upgrade on an older version on my Dell laptop a few months ago so it should work in principal.

**Comment by dtzWill** [comment]
Thanks all! Gave this a go, generally looks good and UEFI situation same as @moredread reports above-- best I could figure out was to just copy the things over (and remove after).  Finally updated my laptop's BIOS xD.

**Pull request merged**